### PR TITLE
12346 - Fix Rubocop Rails/Pluck and Rails/PluckInWhere

### LINKED ISSUE #12346: Rubocop Rails/Pluck and Rails/PluckInWhere
Part of #11482

- [ ] [Rails/Pluck](https://www.rubydoc.info/gems/rubocop-rails/2.9.1/RuboCop/Cop/Rails/Pluck)
- [ ] [Rails/PluckInWhere](https://www.rubydoc.info/gems/rubocop-rails/2.8.1/RuboCop/Cop/Rails/PluckInWhere)

### PR DESCRIPTION
**Pull request by chahmedejaz**
#### What? Why?

- Closes #12346 
-  This PR fixes the `Rails/Pluck` and `Rails/PluckInWhere` Rubocop errors

#### What should we test?

- Build should be green

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

### CODE REVIEW

File: spec/controllers/admin/bulk_line_items_controller_spec.rb [hunk 1]
```diff
@@ -59,15 +59,13 @@
...
+                 }).to eq(true)
         end
 
         it "returns distributor object with id key" do
-          expect(json_response['line_items'].map{ |line_item|
-                   line_item['supplier']
-                 }.all?{ |d| d.key?('id') }).to eq(true)
+          expect(json_response['line_items'].pluck(:supplier).all?{ |d| d.key?('id') }).to eq(true)
         end
       end
```
Comment (dacook) [inline spec/controllers/admin/bulk_line_items_controller_spec.rb:70]: I reckon these expectiations could be refactored even more..

File: spec/controllers/api/v0/orders_controller_spec.rb [hunk 2]
```diff
@@ -161,19 +161,17 @@ module Api
...
 
             expect(json_response['orders']
-              .map{ |o| o[:id] }).to match_array serialized_orders([order2, order3, order1, order4,
-                                                                    order7])
-                .map{ |o|
-                                                   o["id"]
-                                                 }
+              .pluck(:id)).to match_array serialized_orders([order2, order3, order1, order4,
+                                                             order7])
+                .pluck(:id)
```
Comment (dacook) [inline spec/controllers/api/v0/orders_controller_spec.rb]: This is a bit confusing, and probably could be appended to the previous line.

Comment (dacook) [reply spec/controllers/api/v0/orders_controller_spec.rb]: I would try to put the `expect()` on one line, and the `to ...` on the next line.
(also on next one below(

**Review by dacook — CHANGES_REQUESTED** [review]
Nice cleanup. I have a suggestion for cleaning up a little bit more ;)

File: spec/controllers/admin/bulk_line_items_controller_spec.rb (hunk 1 above)
Comment (chahmedejaz) [reply spec/controllers/admin/bulk_line_items_controller_spec.rb]: It's been addressed here:
https://github.com/openfoodfoundation/openfoodnetwork/pull/12599/commits/ff10a36e4a4b2077c9549511c25e9fe1903f8d09

Comment (chahmedejaz) [reply spec/controllers/admin/bulk_line_items_controller_spec.rb]: It's been fixed here 🙂: https://github.com/openfoodfoundation/openfoodnetwork/pull/12599/commits/ff10a36e4a4b2077c9549511c25e9fe1903f8d09

File: spec/controllers/api/v0/orders_controller_spec.rb (hunk 2 above)
Comment (chahmedejaz) [reply spec/controllers/api/v0/orders_controller_spec.rb]: It's been fixed here:
https://github.com/openfoodfoundation/openfoodnetwork/pull/12599/commits/72640cfc2ac2d047e8dc657acc2b610b8b5e35ab

I've created a common memoized variable and it eliminated the need for multiline.

**Review by dacook — APPROVED** [review]
Perfect, great job! 💯

**Comment by dacook** [comment]
Just needs a conflict fix and we're good to go.

**Comment by chahmedejaz** [comment]
Thanks, David. It's resolved :)

**Pull request merged**